### PR TITLE
Notifi Client implementation

### PR DESCRIPTION
**Pull request by nubstick27**
Rather than separate hooks, it would be easier to wrap a class which exposes the functionality. Taking a step in that direction by declaring this new service in a new `core` package, with an implementation in `axios-adapter` package.

The service is exposed to `react-hooks` via a new `useNotifiService` hook instead.

### CODE REVIEW

File: packages/notifi-axios-adapter/lib/mutations/createAlertImpl.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+    targetGroup {
+      id
+      name
+    }
+  }
+}
+`.trim();
+
+const createAlertImpl = makeRequest<CreateAlertInput, CreateAlertResult>(
```
Comment (nimesh-irisio) [inline packages/notifi-axios-adapter/lib/mutations/createAlertImpl.ts:29]: on error, what's the expected pattern consumers should use? As in, will makeRequest throw, or would the entire payload be returned, and the consumer should check for the error field?

**Review by nimesh-irisio — APPROVED** [review]
Really liking this so far. Only question is around the error handling pattern

File: packages/notifi-axios-adapter/lib/GqlError.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export default class GqlError extends Error {
```
Comment (nimesh-irisio) [inline packages/notifi-axios-adapter/lib/GqlError.ts:24]: Are these only for app level errors or for generic?
app being 200 status code, but errors field populated and no result

**Comment by nubstick27** [comment]
The encouraged usage pattern is now to use `useNotifiClient` hook, which exposes a "fat" client with encapsulated functionality.

File: packages/notifi-axios-adapter/lib/mutations/createAlertImpl.ts (hunk 1 above)
Comment (nubstick27) [reply packages/notifi-axios-adapter/lib/mutations/createAlertImpl.ts:29]: In the new design, things are wrapped into a "fat" client which will capture some errors, but they should still handle things that are bubbled up to them

File: packages/notifi-axios-adapter/lib/GqlError.ts (hunk 2 above)
Comment (nubstick27) [reply packages/notifi-axios-adapter/lib/GqlError.ts:24]: Uhh I think these are for app errors.

File: packages/notifi-react-hooks/lib/hooks/index.ts [hunk 3]
```diff
@@ -1,29 +1,4 @@
...
-  useCreateTelegramTarget,
-  useGetFilters,
-  useGetSourceGroups,
-  useGetTargetGroups,
-  useLoginFromDao,
-  useNotifiConfig,
-  useNotifiJwt,
-  useUpdateTargetGroup
-};
+export { BlockchainEnvironment, useNotifiClient, MessageSigner };
```
Comment (nubstick27) [inline packages/notifi-react-hooks/lib/hooks/index.ts]: big public API changes here

File: packages/notifi-react-hooks/lib/hooks/useNotifiClient.ts [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+  const newTarget = await service.createTelegramTarget({
+    name: telegramId,
+    value: telegramId
+  });
+  return newTarget.id;
+};
+
+const useNotifiClient = (
+  env = BlockchainEnvironment.MainNetBeta,
+  signer: MessageSigner
```
Comment (nubstick27) [inline packages/notifi-react-hooks/lib/hooks/useNotifiClient.ts]: on second thought, maybe better to expose this dependency as a paramter of `logIn`

File: packages/notifi-axios-adapter/lib/NotifiAxiosService.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import logInFromDaoImpl from './mutations/logInFromDaoImpl';
+import updateTargetGroupImpl from './mutations/updateTargetGroupImpl';
+
+export type NotifiAxiosServiceConfig = Readonly<{
+  gqlUrl: string;
+  getJwt: () => string | undefined;
+}>;
+
+export class NotifiAxiosService implements NotifiService {
+  createAlert: NotifiService['createAlert'];
```
Comment (nimesh-irisio) [inline packages/notifi-axios-adapter/lib/NotifiAxiosService.ts:24]: Is single flat namespace scalable when/if our API grows significantly?

**Review by nimesh-irisio — APPROVED** [review]

File: packages/notifi-axios-adapter/lib/NotifiAxiosService.ts (hunk 5 above)
Comment (nubstick27) [reply packages/notifi-axios-adapter/lib/NotifiAxiosService.ts:24]: My guess is "probably" -- there will be nuanced cross-dependencies that would make things easier to export as a single implementation.

FWIW, the core interfaces are all single-operation interfaces, so we can reconsider very agile-ly later.

File: packages/notifi-react-hooks/lib/hooks/useNotifiClient.ts [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import useNotifiJwt from './useNotifiJwt';
+
+export type NotifiClientConfig = Readonly<{
+  daoAddress: string;
+  walletPublicKey: string;
+  env?: BlockchainEnvironment;
+}>;
+
+export class NotifiClientError extends Error {
```
Comment (nimesh-irisio) [inline packages/notifi-react-hooks/lib/hooks/useNotifiClient.ts:26]: Is this the class we are going to use for GQL errors where we get a JSON blob back in the error field?

Comment (nubstick27) [reply packages/notifi-react-hooks/lib/hooks/useNotifiClient.ts:26]: No, this is for unknown errors. I've re-exported the GqlError class from axios-adapter in the latest commit. The intent is to provide type safety around the `error` that's returned in the hook, but clients can still catch the underlying thing directly

the pattern is something like
```
try { ... } catch (e: unknown) {
  if (e instanceof Error) {
    setError(e);
  } else {
    setError(new NotifiClientError(e));
  }
  throw e; // rethrow original
}
```

**Pull request merged**